bundler/spec/bundler/ruby_dsl_spec.rb

```diff
@@ -97,53 +99,67 @@ class MockDSL
     end
 
     context "with a file option" do
-      let(:options) { { :file => "foo" } }
-      let(:version) { "3.2.2" }
-      let(:ruby_version) { "3.2.2" }
+      let(:file) { ".ruby-version" }
+      let(:options) do
+        { :file => file,
+          :patchlevel => patchlevel,
+          :engine => engine,
+          :engine_version => engine_version }
+      end
       let(:ruby_version_arg) { nil }
-      let(:engine_version) { version }
-      let(:patchlevel) { nil }
-      let(:engine) { "ruby" }
-      let(:project_root) { Pathname.new("/path/to/project") }
+      let(:file_content) { "#{version}\n" }
 
       before do
-        allow(Bundler).to receive(:read_file).with(project_root.join("foo")).and_return("#{version}\n")
-        allow(Bundler).to receive(:root).and_return(Pathname.new("/path/to/project"))
+        allow(Bundler).to receive(:read_file).with(project_root.join(file)).and_return(file_content)
       end
 
       it_behaves_like "it stores the ruby version"
 
+      context "with the ruby- prefix in the file" do
+        let(:file_content) { "ruby-#{version}\n" }
+
+        it_behaves_like "it stores the ruby version"
+      end
+
       context "and a version" do
-        let(:ruby_version_arg) { "2.0.0" }
+        let(:ruby_version_arg) { version }
 
         it "raises an error" do
-          expect { subject }.to raise_error(Bundler::GemfileError, "Cannot specify version when using the file option")
+          expect { subject }.to raise_error(Bundler::GemfileError, "Do not pass version argument when using :file option")
         end
       end
-    end
 
-    context "with a (.tool-versions) file option" do
-      let(:options) { { :file => "foo" } }
-      let(:version) { "3.2.2" }
-      let(:ruby_version) { "3.2.2" }
-      let(:ruby_version_arg) { nil }
-      let(:engine_version) { version }
-      let(:patchlevel) { nil }
-      let(:engine) { "ruby" }
-      let(:project_root) { Pathname.new("/path/to/project") }
+      context "with a @gemset" do
+        let(:file_content) { "ruby-#{version}@gemset\n" }
 
-      before do
-        allow(Bundler).to receive(:read_file).with(project_root.join("foo")).and_return("nodejs 18.16.0\nruby #{version} # This is a comment\npnpm 8.6.12\n")
-        allow(Bundler).to receive(:root).and_return(Pathname.new("/path/to/project"))
+        it "raises an error" do
+          expect { subject }.to raise_error(Gem::Requirement::BadRequirementError, "Illformed requirement [\"#{version}@gemset\"]")
```

Comment:
I covered the reason for rejecting `@gemset` in the linked ticket. TL;DR rvm says it's incompatible with other version managers and you should use `.ruby-gemset`.